### ISSUE TITLE
BitCount flag

### ISSUE THREAD

**Issue by SporyTike** [RetroAchievements/RAIntegration#529]
Measured is a very cool feature to see how far your achievement progress is but it has a small flaw. If you want to create an achievement with a progress bar which displays the progress correctly for something like collectibles which are saved in bits, you need to use AddSource and multiple lines which sum the bits together. That's a problem if you have many collectibles (and maybe multiple save files).

A good example is this achievement:
http://retroachievements.org/achievement/83940

It's annoying to code tons of AddSource lines together even with the use of RATools and in such cases it can also reach the limits of the toolkit. That's why I would like to have a BitCount flag which counts the active bits of a memory and gives the number to the following line.

As example I have this:
Mem 0x1234 has Value 0x68 (0b0**11**0**1**000)
Mem 0x4321 has Value 0x73 (0b0**111**00**11**)

`1 BitCount Mem 8-bit 0x1234`
`2 BitCount Mem 8-bit 0x4321`
`3 ⠀⠀⠀ ⠀⠀⠀ Value ⠀ ⠀ 0x00⠀⠀ = Value ⠀ ⠀ 0x08`

The first condition reads the active bits in 0x1234 and gets the number 3 which it hands over to the second condition. The active bit counter for 0x4321 is 5 so it takes the 3 from the first line and adds 5 to it. The sum would be 8 and would be put on the value in the third condition.

This would make the logic for achievements like this much smaller because only three lines are needed instead of the 16 lines which are needed at the moment. And in the most cases the number of lines is the double because there is a Delta check as save protection.

This feature could be linked with Measured to get the current number of active bits  as the progress value.

**Comment by Keltron3030** [comment]
See #524

**Comment by Jamiras** [comment]
Closing as solved by #524.

After [discussion in discord](https://discordapp.com/channels/310192285306454017/310195377993416714/701135985878237225), it seems like the best way to implement this is as a Size, so your example would be:
```
1 AddSource Mem BitCount 0x1234
2           Mem BitCount 0x4321 = Value ⠀ ⠀ 0x08
```

Using a Size allows for all the normal modifiers: Delta, Prior, AddHits, ResetIf, etc. The only limitation is that it only works one byte at a time. As it's effectively shorthand for 8 AddSources, this seems like a reasonable compromise. And you can AddSource multiple BitCounts for sizes greater than a byte.
